sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -112,8 +105,9 @@ private[sql] object FileSourceStrategy extends Strategy with Logging {
 
       val readFile = files.fileFormat.buildReader(
         sqlContext = files.sqlContext,
+        physicalSchema = files.dataSchema,
```

Comment:
why do we call it `physicalSchema`? It's kind of a reconciled schema that may not equal to the real physical schema.